lib/setupmanagers/qemuhck/storage_manager.rb

```diff
@@ -0,0 +1,116 @@
+require_relative '../../auxiliary/string_helper'
+
+# AutoHCK module
+module AutoHCK
+  # QemuMachine class
+  class QemuMachine
+    # StorageManager class
+    class StorageManager
+      include Helper
+
+      CONFIG_JSON = 'lib/setupmanagers/qemuhck/network_manager.json'
+      IMAGE_FORMAT = 'qcow2'
+
+      def initialize(id, client_id, qemu_config, qemu_options, logger)
+        @id = id
+        @client_id = client_id
+        @logger = logger
+
+        @workspace_path = qemu_options['workspace_path']
+        @qemu_img_bin = qemu_config['qemu_img_bin']
+        @fs_test_image = qemu_config['fs_test_image']
+
+        @base_image_path = Pathname.new(qemu_config['images_path']).join(qemu_options['image_name'])
+        @test_image_path = Pathname.new(@workspace_path).join("client#{@client_id}_test_image.#{IMAGE_FORMAT}")
+      end
+
+      def read_device(device)
+        @logger.info("Loading device: #{device}")
+        device_json = "#{QemuMachine::DEVICES_JSON_DIR}/#{device}.json"
+        unless File.exist?(device_json)
+          @logger.fatal("#{device} does not exist")
+          raise(InvalidConfigFile, "#{device} does not exist")
+        end
+        Json.read_json(device_json, @logger)
+      end
+
+      def device_command_info(type, device_name, command_options, qemu_replacement_list)
+        device = read_device(device_name)
+        replacement_list = qemu_replacement_list.merge command_options
+        device_command = replace_string_recursive(device['command_line'].join(' '), replacement_list)
+
+        @logger.debug("Device #{device_name} used as #{type} device")
+        @logger.debug("Device command: #{device_command}")
+
+        device_command
+      end
+
+      def snapshot_path
+        filename = File.basename(@base_image_path, '.*')
+        "#{@workspace_path}/#{filename}-snapshot.#{IMAGE_FORMAT}"
+      end
+
+      def check_image_exist
+        File.exist?(@base_image_path)
+      end
+
+      def create_image(path, size_gb)
+        run_cmd("#{@qemu_img_bin} create -f #{IMAGE_FORMAT} #{path} #{size_gb}G")
+      end
+
+      def create_boot_image
+        create_image(@base_image_path, 150)
+      end
+
+      def create_test_image
+        if File.exist?(@fs_test_image)
+          @logger.info("Coping test image for CL#{@client_id}")
+          File.copy(@fs_test_image, @test_image_path)
+        else
+          @logger.info("Creating CL#{@client_id} test image")
+          create_image(@test_image_path, 30)
+        end
+      end
+
+      def create_boot_snapshot
+        @logger.info("Creating CL#{@client_id} snapshot file")
+        run_cmd("#{@qemu_img_bin} create -f #{IMAGE_FORMAT} -F #{IMAGE_FORMAT} -b #{@base_image_path} #{snapshot_path}")
+      end
+
+      def delete_boot_snapshot
+        FileUtils.rm_f(snapshot_path)
+      end
+
+      def boot_device_command(device_name, run_opts, qemu_replacement_list = {})
+        type = __method__.to_s.split('_').first
```

Comment:
This is excessively complex. I understand it is here to remove the duplication of the word `boot`, but it introduces a duplication of `_` instead and moreover introduces some reflective programming construct and string manipulation. Overall it's not reducing but adding complexity.

I know it's from `NetworkManager` but I don't want another copies of this pattern. This variable is only used for a debug log in this class so I suggest to simply remove it.